### PR TITLE
chore: update README and CONTRIBUTING guides

### PR DESCRIPTION
**Pull request by jedevc**
This patch performs several updates to the internal docs.

Notably:
- Links CONTRIBUTING.md from the README.md
- Moves the content from TESTING.md into CONTRIBUTING.md
- Several other minor readability improvements

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -62,9 +62,9 @@ If there are code changes in the SDKs, run `changie new` in the corresponding di
 
 Remember to add & commit the release notes fragment.
 This will be used at release time, in the changelog.
-Here is an example of the end-result for all release notes fragments: https://github.com/dagger/dagger/blob/v0.6.4/.changes/v0.6.4.md
+Here is an example of the end-result for all release notes fragments: <https://github.com/dagger/dagger/blob/v0.6.4/.changes/v0.6.4.md>
```
Comment (TomChv) [inline CONTRIBUTING.md:65]: Link already work, why add these `<`, `>`? Is it for readability improvement?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -200,6 +200,48 @@ If there isn't a newer version to upgrade to, we'll be in a tougher spot and may
 
 ## FAQ
 
+### How to run a development engine?
```
Comment (TomChv) [inline CONTRIBUTING.md:203]: Great addition! I agree that this should be in this file instead of testing.

**Review by gerhard — APPROVED** [review]
Thank you for all these improvements!

File: CONTRIBUTING.md (hunk 1 above)
Comment (gerhard) [reply CONTRIBUTING.md:65]: Forces a URL in Markdown: https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#links

**Pull request merged**